build/files.c

```diff
@@ -877,13 +887,29 @@ static VFA_t const virtualAttrs[] = {
  */
 static rpmRC parseForSimple(char * buf, FileEntry cur, ARGV_t * fileNames)
 {
-    char *s, *t;
+    char *s, *t, *end;
+    char *delim = " \t\n";
+    int quotes = 0;
     rpmRC res = RPMRC_OK;
     int allow_relative = (RPMFILE_PUBKEY|RPMFILE_DOC|RPMFILE_LICENSE);
 
     t = buf;
-    while ((s = strtokWithQuotes(t, " \t\n")) != NULL) {
+    while ((s = strtokWithQuotes(t, delim, &quotes)) != NULL) {
 	t = NULL;
+	end = s + strlen(s) - 1;
+
+	/* Syntax checks */
+	if (quotes == 1) {
+	    rpmlog(RPMLOG_ERR, _("Missing quote: %s\n"), s);
```

Comment:
I was about to say this should report a line number, but the existing error message doesn't have that either and the needed context is missing multiple layers of calls above, so lets not go there now. This series is long enough as it is :laughing:

Reply:
Yup, a line number would be useful of course, but let's leave that for another time.